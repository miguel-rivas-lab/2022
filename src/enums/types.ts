import i18n from "../i18n";

export enum typeEnum {
  userFlow = 100,
  wireFrame,
  app,
  prototype,
  landingPage,
  motionGraphics,
  animation3D,
  document,
  module,
  socialMedia,
  graphicDesign,
  group,
  location,
}

export const type = {
  userFlow: i18n.t("enum.projectsType.userFlow"),
  wireFrame: i18n.t("enum.projectsType.wireFrame"),
  app: i18n.t("enum.projectsType.app"),
  prototype: i18n.t("enum.projectsType.prototype"),
  landingPage: i18n.t("enum.projectsType.landingPage"),
  motionGraphics: i18n.t("enum.projectsType.motionGraphics"),
  animation3D: i18n.t("enum.projectsType.animation3D"),
  document: i18n.t("enum.projectsType.document"),
  module: i18n.t("enum.projectsType.module"),
  socialMedia: i18n.t("enum.projectsType.socialMedia"),
  graphicDesign: i18n.t("enum.projectsType.graphicDesign"),
  group: i18n.t("enum.projectsType.group"),
  location: i18n.t("enum.projectsType.location"),
};