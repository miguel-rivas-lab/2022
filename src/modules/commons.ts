import Vue from 'vue';
import NanoRow from 'nano-grid/vue/row.vue';
import NanoColumn from 'nano-grid/vue/column.vue';
import NanoIcon from 'nano-grid/vue/icon.vue';
import NanoBtn from 'nano-grid/vue/btn.vue';
import NanoScrollArea from 'nano-grid/vue/scroll-area.vue';
import NumberInput from 'nano-grid/vue/number-input.vue';

import Container from '../components/container.vue';

Vue.component("container", Container);

Vue.component("row", NanoRow);
Vue.component("column", NanoColumn);
Vue.component("icon", NanoIcon);
Vue.component("btn", NanoBtn);
Vue.component("scroll-area", NanoScrollArea);
Vue.component("number-input", NumberInput);

// ---------------------------------- Alias:Icon
Vue.component("m-icon", Vue.extend({
  props: {
    mode: {
      default: 'mdi'
    }
  },
  extends: NanoIcon
}));
// ---------------------------------- Alias:Column
Vue.component("prefix", Vue.extend({
  props: {
    mode: {
      default: 'prefix'
    }
  },
  extends: NanoColumn
}));

Vue.component("suffix", Vue.extend({
  props: {
    mode: {
      default: 'suffix'
    }
  },
  extends: NanoColumn
}));

Vue.component("t-column", Vue.extend({
  props: {
    tableElement: {
      default: true
    }
  },
  extends: NanoColumn
}));

Vue.component("t-prefix", Vue.extend({
  props: {
    mode: {
      default: 'prefix'
    },
    tableElement: {
      default: true
    }
  },
  extends: NanoColumn
}));

Vue.component("t-suffix", Vue.extend({
  props: {
    mode: {
      default: 'suffix'
    },
    tableElement: {
      default: true
    }
  },
  extends: NanoColumn
}));

// ---------------------------------- Alias:Row
Vue.component("t-row", Vue.extend({
  props: {
    tableElement: {
      default: true
    }
  },
  extends: NanoRow
}));