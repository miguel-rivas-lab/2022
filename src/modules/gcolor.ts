export class gColor {
  label: string;
  hex: string;
  item?: string;

  constructor(label: string, hex: string, item?: string) {
    this.label = label;
    this.hex = hex;
    this.item = item;
  }

  get RGB() {
    return {
      red: parseInt(this.hex.substr(1, 2), 16),
      green: parseInt(this.hex.substr(3, 2), 16),
      blue: parseInt(this.hex.substr(5, 2), 16),
    }
  }

  get HSL() {
    const r = this.RGB.red / 255,
      g = this.RGB.green / 255,
      b = this.RGB.blue / 255,
      cmin = Math.min(r, g, b),
      cmax = Math.max(r, g, b),
      delta = cmax - cmin;

    let h = 0,
      s = 0,
      l = 0;

    if (delta === 0) {
      h = 0;
    } else if (cmax === r) {
      h = ((g - b) / delta) % 6;
    } else if (cmax === g) {
      h = (b - r) / delta + 2;
    } else {
      h = (r - g) / delta + 4;
    }

    h = Math.round(h * 60);

    if (h < 0) {
      h += 360;
    }

    l = (cmax + cmin) / 2;
    s = delta === 0 ? 0 : delta / (1 - Math.abs(2 * l - 1));

    s = +(s * 100).toFixed(1);
    l = +(l * 100).toFixed(1);

    return { hue: h, saturation: s, lightness: l };
  }

  get hsl() {
    return `hsl(${this.HSL.hue}, ${this.HSL.saturation}%, ${this.HSL.lightness}%)`;
  }

  get rgb() {
    return `rgb(${this.RGB.red}, ${this.RGB.green}, ${this.RGB.blue})`;
  }

  get spinalCase() {
    return this.label
      .replace(/[!,.*+?^${(|[\\})\]:]/g, '')
      .replace(/&/g, 'and')
      .trim()
      .replace(/[ ]{1,}/g, "-")
      .toLowerCase();
  }
}