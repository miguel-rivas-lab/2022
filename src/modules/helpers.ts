import { DateTime } from "luxon";

const helpers = {
  dateToNumber(date: string): number {
    return parseInt(date.split("/").join(""));
  },
  turingDate(date) {
    const newDate = DateTime.fromISO(date.replace(/[/]/g, "-")).minus({years: 1990, months: 5, days: 5});
    return `Y${newDate.year} M${newDate.month} D${newDate.day}`;
  },
  getID(client: string, date: string): string {
    client = client.replace(/[\s./]/g, "").toLowerCase();
    date = date.replace(/\//g, "");
    return `${client}_${date}`;
  },
  getNewID(client: string, date: string): string {
    client = client.replace(/[\s./]/g, "").toLowerCase();
    date = date.replace(/\//g, "");
    return `${date}_${client}`;
  },
}

export default helpers;