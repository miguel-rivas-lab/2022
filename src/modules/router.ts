import Vue from 'vue';
import VueRouter, { RouteConfig } from 'vue-router';

import { firebaseApp } from "./firebase";
import "firebase/auth";

import ColorsPanelView from '../views/colors-panel.vue';
import ColorsWorkareaView from '../views/colors-workarea.vue';
import CubeWorkareaView from '../views/cube-workarea.vue';
import CubePanelView from '../views/cube-panel.vue';
import GridWorkareaView from '../views/grid-workarea.vue';
import GridPanelView from '../views/grid-panel.vue';
import HomeWorkareaView from '../views/home-workarea.vue';
import HomePanelView from '../views/home-panel.vue';
import ProjectsWorkareaView from '../views/projects-workarea.vue';
import ProjectsPanelView from '../views/projects-panel.vue';
import StatisticsWorkareaView from '../views/statistics-workarea.vue';
import GearWorkareaView from '../views/gear-workarea.vue';
import GearPanelView from '../views/gear-panel.vue';
import WheelWorkareaView from '../views/wheel-workarea.vue';
import WheelPanelView from '../views/wheel-panel.vue';
import LocationsPanelView from '../views/locations-panel.vue';
import LocationsWorkareaView from '../views/locations-workarea.vue';
import CryptoWorkareaView from '../views/crypto-workarea.vue';

import Login from '../views/login-workarea.vue';

import DefaultPanelView from '../views/default-panel.vue';

Vue.use(VueRouter);

const routes: Array<RouteConfig> = [
  {
    name: 'home',
    path: '/',
    components: {
      workarea: HomeWorkareaView,
      panel: HomePanelView,
    },
  },
  {
    name: 'projects',
    path: '/projects',
    components: {
      workarea: ProjectsWorkareaView,
      panel: ProjectsPanelView,
    },
  },
  {
    name: 'colors',
    path: '/colors',
    components: {
      workarea: ColorsWorkareaView,
      panel: ColorsPanelView,
    },
  },
  {
    name: 'cube',
    path: '/cube',
    components: {
      workarea: CubeWorkareaView,
      panel: CubePanelView,
    },
  },
  {
    name: 'crypto',
    path: '/crypto',
    components: {
      workarea: CryptoWorkareaView,
      panel: DefaultPanelView,
    },
    meta: { requiresAuth: true }
  },
  {
    name: 'login',
    path: '/login',
    components: {
      workarea: Login,
      panel: DefaultPanelView,
    },
  },
  {
    name: 'statistics',
    path: '/statistics',
    components: {
      workarea: StatisticsWorkareaView,
      panel: DefaultPanelView,
    },
  },
  {
    name: 'grid',
    path: '/grid',
    components: {
      workarea: GridWorkareaView,
      panel: GridPanelView,
    },
  },
  {
    name: 'gear',
    path: '/gear',
    components: {
      workarea: GearWorkareaView,
      panel: GearPanelView,
    },
  },
  {
    name: 'locations',
    path: '/locations',
    components: {
      workarea: LocationsWorkareaView,
      panel: LocationsPanelView,
    },
  },
  {
    name: 'wheel',
    path: '/wheel',
    components: {
      workarea: WheelWorkareaView,
      panel: WheelPanelView,
    },
  },
  {
    path: '*',
    redirect: '/'
  },
];

const router = new VueRouter({
  mode: 'history',
  base: process.env.BASE_URL,
  routes,
})

router.beforeEach((to, from, next) => {
  const requiresAuth = to.matched.some(record => record.meta.requiresAuth);
  const isAuthenticated = firebaseApp.auth().currentUser;
  if (requiresAuth && !isAuthenticated) {
    next("/login");
  } else {
    next();
  }
});

export default router;