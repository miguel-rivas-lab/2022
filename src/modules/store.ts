import Vue from 'vue';
import Vuex from 'vuex';

Vue.use(Vuex);

export const store = new Vuex.Store({
  state: {
    panel: true,
    panelSize: ['300', '100%-350'],
    theme: false,
    parallelUniverse: false,
    alert: {
      message: "",
      status: "",
    },
    modal: {
      visible: false,
    },
    selection: {
      login: {
        email: "",
        password: "",
        logged: false,
      },
      locations: {
        mapCenter: {
          lat: 31,
          lng: -94,
        },
        mapZoom: 4,
        viewPath: false,
      },
      projects: {
        filterData: 'all',
        currentLink: {
          src: "",
          frameborder: "0",
          allowfullscreen: true
        },
      },
      home: {
        sceneRotation: 1,
        cover: true,
        rotateX: false,
        rotateY: false,
        rotateZ: true,
        grid: true,
      },
      grid: {
        row: "Row",
        integrate: false,
        breakpoint: "-",
        spacing: 0,
        columns: [],
      },
      cube: {
        cubeRotation: 3,
        pause: false,
        grid: true,
        lines: false,
      },
      gear: {
        cornerAmount: 40,
        outterCircleRadius: 50,
        innerCircleRadius: 45,
        mainPerforationRadius: 5,
        sidePerforationAmount: 4,
        sidePerforationRadius: 12,
        sidePerforationDistance: 25,
      },
      wheel: {
        colors: 12,
      }
    },
  },
  mutations: {
    setValue(state, payload) {
      state[payload.name] = payload.value;
    },
    setProject(state, payload) {
      state.selection.projects.currentLink.src = payload.value;
    },
    centerMap(state, payload) {
      state.selection.locations.mapCenter = payload.position;
      state.selection.locations.mapZoom = payload.zoom;
    },
    toggleValue(state, payload) {
      state[payload] = !state[payload];
    },
    addColumn(state, payload) {
      state.selection.grid.columns.push(payload);
    },
    removeColumn(state, index) {
      state.selection.grid.columns.splice(index, 1);
    },
  },
  // actions: {},
  // modules: {},
  getters: {
    getPanelVisibility: state => state.panel,
    getPanelSize: state => state.panelSize,
    getTheme: state => state.theme,
    getUniverse: state => state.parallelUniverse,
    getAlert: state => state.alert,
    getModal: state => state.modal,
    getHomeSelection: state => state.selection.home,
    getGridSelection: state => state.selection.grid,
    getCubeSelection: state => state.selection.cube,
    getGearSelection: state => state.selection.gear,
    getWheelSelection: state => state.selection.wheel,
    getLocationSelection: state => state.selection.locations,
    getLoginSelection: state => state.selection.login,
    getFilterData: state => state.selection.projects,
    
  }
});