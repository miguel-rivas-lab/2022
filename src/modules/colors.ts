import { gColor } from "./gcolor";

export const gColorsDB = [
  new gColor('Shamrock', '#34ce7e'),
  new gColor('Emerald', '#5acf57'),
  new gColor('Laurel', '#328538'),
  new gColor('Green Pea', '#20723b'),
  new gColor('Elf Green', '#198663'),
  new gColor('Lake', '#257d93'),
  new gColor('Mariner', '#3876cc'),
  new gColor('Denim', '#116ec4'),
  new gColor('Cobalt Blue', '#0B4EAB'),
  new gColor('Royal Purple', '#64429c'),
  new gColor('Purple Heart', '#7d32b0'),
  new gColor('Blue Violet', '#994aed'),
  new gColor('Fuchsia', '#AF08AA'),
  new gColor('Razzmatazz', '#da0040'),
  new gColor('Alizarin', '#e72113'),
  new gColor('Fire Brick', '#a5213a'),
  new gColor('Persian Red', '#d73333'),
  new gColor('Blush', '#bf4a67'),
  new gColor('Desert', '#a0613d'),
  new gColor('Rust', '#c84d11'),
  new gColor('Sepia', '#704d16'),
  new gColor('Golden Brown', '#9e6710'),
  new gColor('Yukon Gold', '#8c741d'),
  new gColor('Burn Orange', '#f9722e'),
  new gColor('Sunglow', '#ffcf25'),
  new gColor('Gold Tips', '#e6b031'),
  new gColor('Milk Punch', '#f4e7c4'),
  new gColor('Silver', '#c0c0c0'),
  new gColor('Gravel', '#6e6e6e'),
  new gColor('Charcoal', '#444444'),
  new gColor('Cod Grey', '#1b1b1b'),
];