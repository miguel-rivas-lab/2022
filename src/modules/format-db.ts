import helpers from "mr-kernel/modules/helpers";

// ---------------- Enums

import { tool, toolEnum } from "mr-kernel/enums/tools";
import { role, roleEnum } from "mr-kernel/enums/roles";
import { type, typeEnum } from "mr-kernel/enums/types";
import { client, clientEnum } from "mr-kernel/enums/clients";

import { ProjectFirebase } from "mr-kernel/interfaces/project-firebase";

interface ProjectLinkComb {
  url: string;
  text: string;
  params?: Array<string>;
  self?: boolean;
}

export interface ProjectComb {
  date: string;
  title: string;
  types?: Array<string> | Array<number>;
  roles?: Array<string> | Array<number>;
  clients: Array<string> | Array<number>;
  tools?: Array<string> | Array<number>;
  links?: Array<ProjectLinkComb>;
  disabled?: boolean;
  children?: Array<string>;
}

export function formatRawDBToFirebase(item: ProjectFirebase): ProjectFirebase {
  let links = [];
  if (item.links?.length > 0) {
    links = item.links.map(
      link => {
        const params = link.params ? link.params : [];
        return {
          "url": link.url,
          "text": link.text,
          "params": params,
          "self": link.self || false,
        }
      }
    );
  }

  const roles = item.roles.sort();
  const tools = item.tools.sort();
  const clients = item.clients.sort();
  const types = item.types.sort();
  const disabled = item.disabled ? true : false;
  const children = item.children || [];

  return {
    "title": item.title,
    "clients": clients,
    "date": item.date,
    "types": types,
    "disabled": disabled,
    "links": links,
    "roles": roles,
    "tools": tools,
    "children": children,
  }
}

export function formatFirebaseDBToJSON(querySnapshot) {
  const projectsDB = {};
  querySnapshot.docs.forEach(doc => {
    const p = doc.data();

    const links = p.links.map(
      link => {
        const params = link.params.length ? `?${link.params.join("&")}` : '';
        const url = link.url;

        return {
          "url": `${url}${params}`,
          "text": link.text,
          "self": link.self,
        }
      }
    );

    const roles = p.roles.map(
      item => role[roleEnum[item]]
    );

    const tools = p.tools.map(
      item => tool[toolEnum[item]]
    );

    const clients = p.clients.map(
      item => client[clientEnum[item]]
    );

    const types = p.types.map(
      item => type[typeEnum[item]]
    ).join(' & ');

    const project = {
      "title": p.title,
      "clients": clients,
      "date": p.date,
      "types": types,
      "disabled": p.disabled,
      "links": links,
      "roles": roles,
      "tools": tools,
      "children": p.children,
      "image": "",
    };

    const id = helpers.getNewID(project.clients[0], project.date);

    try {
      project.image = `https://miguel-rivas.github.io/zapp/img/preview-wide/${id}.jpg`;
    }
    catch {
      project.image = require(`@/img/miguelrivas.jpg`);
    }
    projectsDB[id] = project;
  });
  return projectsDB;
}

export function formatRawDBToJSON(item) {
  const result = {};
  item.forEach(doc => {
    const p = doc;

    const links = p.links.map(
      link => {
        const params = link.params.length ? `?${link.params.join("&")}` : '';
        const url = link.url;

        return {
          "url": `${url}${params}`,
          "text": link.text,
          "self": link.self,
        }
      }
    );

    const roles = p.roles.map(
      item => role[roleEnum[item]]
    );

    const tools = p.tools.map(
      item => tool[toolEnum[item]]
    );

    const clients = p.clients.map(
      item => client[clientEnum[item]]
    );

    const types = p.types.map(
      item => type[typeEnum[item]]
    ).join(' & ');

    const project = {
      "title": p.title,
      "clients": clients,
      "date": p.date,
      "types": types,
      "disabled": p.disabled,
      "links": links,
      "roles": roles,
      "tools": tools,
      "children": p.children,
      "image": "",
    };

    const id = helpers.getNewID(project.clients[0], project.date);

    try {
      project.image = `https://miguel-rivas.github.io/zapp/img/preview-wide/${id}.jpg`;
    }
    catch {
      project.image = require(`@/img/miguelrivas.jpg`);
    }
    result[id] = project;
  });
  return result;
}

export function formatGroupsToProjects(groups) {
  const result = {};
  groups.forEach(group => {
    let position = { lat: '', lng: '' };

    if (group.types.includes(typeEnum.location)) {
      position = group.position;
    }

    const tools = group.tools.map(
      item => tool[toolEnum[item]]
    );

    const clients = group.clients.map(
      item => client[clientEnum[item]]
    );

    const types = group.types.map(
      item => type[typeEnum[item]]
    ).join(' & ');

    const project = {
      "title": clients[0],
      "clients": clients,
      "date": group.date,
      "types": types,
      "links": [],
      "disabled": group.disabled,
      "tools": tools,
      "children": group.children,
      "image": "",
      "position": position,
      "description": group.description,
      "list": group.list,
      "zoom": group.zoom,
    };

    const id = helpers.getNewID(project.clients[0], project.date);

    try {
      project.image = `https://miguel-rivas.github.io/zapp/img/preview-wide/${id}.jpg`;
    }
    catch {
      project.image = require(`@/img/miguelrivas.jpg`);
    }
    result[id + "_group"] = project;
  });

  return result;
}

export function sortByDate(a, b) {
  return helpers.dateToNumber(b.date) - helpers.dateToNumber(a.date);
}