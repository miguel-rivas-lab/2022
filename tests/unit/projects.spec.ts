import '@/modules/commons';

import { tool, toolEnum } from "mr-kernel/enums/tools";
import { role, roleEnum } from "mr-kernel/enums/roles";
import { client, clientEnum } from "mr-kernel/enums/clients";
import { type, typeEnum } from "mr-kernel/enums/types";

import { mount } from "@vue/test-utils";
import Gallery from "@/components/gallery.vue";

const users = {
  "name": "Jesus",
  "middleName": "Miguel",
  "lastName": "Rivas",
  "title": "Frontend Developer",
  "website": "miguel-rivas.github.io",
  "location": {
    "city": "Washington",
    "state": "DC"
  },
  "media": {
    "codepen": {
      "user": "planetwurlex"
    },
    "issuu": {
      "user": "jemiguelrivas"
    },
    "linkedin": {
      "user": "jemiguelrivas"
    },
    "github": {
      "user": "jemiguelrivas",
      "development": "miguel-rivas-lab",
      "production": "miguel-rivas"
    }
  }
};

const linkWeb = (item) => `https://${users.media.github.production}.github.io/${item}`;

describe('Gallery component unit tests: ', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(Gallery, {
      propsData: {
        db: [
        {
          "title": "Destapa el Coro",
          "clients": [client[clientEnum.presidente]],
          "types": [type[typeEnum.graphicDesign]],
          "date": "2014/12/05",
          "disabled": false,
          "links": [
            { url: linkWeb("2018/presidente.html"), text: "Presidente" },
          ],
          "roles": [
            role[roleEnum.frontend],
          ],
          "tools": [
            tool[toolEnum.scss],
            tool[toolEnum.javascript],
          ],
          "image": "https://miguel-rivas.github.io/zapp/img/preview.png",
          "children": []
        },
        {
          "title": "Cover",
          "clients": [client[clientEnum.presidente]],
          "types": [type[typeEnum.landingPage]],
          "date": "2021/03/24",
          "disabled": true,
          "links": [],
          "roles": [
            role[roleEnum.frontend],
          ],
          "tools": [
            tool[toolEnum.scss],
            tool[toolEnum.typescript],
          ],
          "image": "https://miguel-rivas.github.io/zapp/img/preview.png",
          "children": []
        }
      ]
      },
    });
    console.log(wrapper.html());
    expect(wrapper.isVueInstance()).toBeTruthy();
    // expect(wrapper.html()).toMatchSnapshot();
  });
});