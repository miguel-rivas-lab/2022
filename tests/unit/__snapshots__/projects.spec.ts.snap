// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery component unit tests:  is a Vue instance 1`] = `
<div class="gallery">
  <div class="container">
    <article class="nano-box">
      <row spacing="100" breakpoint="lg">
        <column size="100%-300"><img loading="lazy" width="1050" height="551" src="https://miguel-rivas.github.io/zapp/img/preview.png" alt="Presidente 2014/12/05"></column>
        <column size="300">
          <h1>Presidente</h1>
          <h2>olis</h2>
          <h3><time>2014/12/05</time></h3>
          <ul class="skills">
            <li>git</li>
          </ul>
          <hr>
          <ul class="navigation"></ul>
        </column>
      </row>
    </article>
  </div>
</div>
`;
